core/src/main/java/org/infinispan/globalstate/ScopedPersistentState.java

```diff
@@ -27,6 +27,26 @@
    String getProperty(String key);
 
    /**
+    * Sets an integer state property.
+    */
+   void setProperty(String key, int value);
+
+   /**
+    * Sets a float state property.
+    */
+   void setProperty(String format, float f);
+
+   /**
+    * Retrieves an integer state property
+    */
+   int getIntProperty(String key);
```

Comment:
If I implemented it that way there would be no way for me to detect T within the method and therefore I could not perform the appropriate string -> type conversion.